code/modules/mining/fulton.dm

```diff
@@ -6,136 +6,179 @@ GLOBAL_LIST_EMPTY(total_extraction_beacons)
-			for(var/turf/open/floor in orange(1, beacon))
-				flooring_near_beacon += floor
-			holder_obj.forceMove(pick(flooring_near_beacon))
-			animate(holder_obj, pixel_z = 10, time = 50)
-			sleep(5 SECONDS)
-			animate(holder_obj, pixel_z = 15, time = 10)
-			sleep(1 SECONDS)
-			animate(holder_obj, pixel_z = 10, time = 10)
-			sleep(1 SECONDS)
-			balloon3 = mutable_appearance('icons/effects/fulton_balloon.dmi', "fulton_retract")
-			balloon3.pixel_y = 10
-			balloon3.appearance_flags = RESET_COLOR | RESET_ALPHA | RESET_TRANSFORM
-			holder_obj.cut_overlay(balloon)
-			holder_obj.add_overlay(balloon3)
-			sleep(0.4 SECONDS)
-			holder_obj.cut_overlay(balloon3)
-			A.set_anchored(FALSE) // An item has to be unanchored to be extracted in the first place.
-			A.set_density(initial(A.density))
-			animate(holder_obj, pixel_z = 0, time = 5)
-			sleep(0.5 SECONDS)
-			A.forceMove(holder_obj.loc)
-			qdel(holder_obj)
-			if(uses_left <= 0)
-				qdel(src)
 
+	balloon_alert_to_viewers("attaching...")
+	playsound(thing, 'sound/items/zip.ogg', vol = 50, vary = TRUE)
+	if(isliving(thing))
+		var/mob/living/creature = thing
+		if(creature.mind)
+			to_chat(thing, span_userdanger("You are being extracted! Stand still to proceed."))
+
+	if(!do_after(user, 5 SECONDS, target = thing))
+		return
+
+	balloon_alert_to_viewers("extracting!")
+	if(loc == user)
+		user.back?.atom_storage?.attempt_insert(src, user, force = STORAGE_SOFT_LOCKED)
+	uses_left--
+
+	if(uses_left <= 0)
+		user.transferItemToLoc(src, thing, TRUE)
+
+	var/mutable_appearance/balloon
+	var/mutable_appearance/balloon2
+	var/mutable_appearance/balloon3
+
+	if(isliving(thing))
+		var/mob/living/creature = thing
+		creature.Paralyze(32 SECONDS) // Keep them from moving during the duration of the extraction
+		if(creature.buckled)
+			creature.buckled.unbuckle_mob(creature, TRUE) // Unbuckle them to prevent anchoring problems
+	else
+		thing.set_anchored(TRUE)
+		thing.set_density(FALSE)
+
+	var/obj/effect/extraction_holder/holder_obj = new(thing.loc)
+	holder_obj.appearance = thing.appearance
+	thing.forceMove(holder_obj)
+	balloon2 = mutable_appearance('icons/effects/fulton_balloon.dmi', "fulton_expand")
+	balloon2.pixel_y = 10
+	balloon2.appearance_flags = RESET_COLOR | RESET_ALPHA | RESET_TRANSFORM
+	holder_obj.add_overlay(balloon2)
+
+	sleep(0.4 SECONDS)
+
+	balloon = mutable_appearance('icons/effects/fulton_balloon.dmi', "fulton_balloon")
+	balloon.pixel_y = 10
+	balloon.appearance_flags = RESET_COLOR | RESET_ALPHA | RESET_TRANSFORM
+	holder_obj.cut_overlay(balloon2)
+	holder_obj.add_overlay(balloon)
+	playsound(holder_obj.loc, 'sound/items/fultext_deploy.ogg', vol = 50, vary = TRUE, extrarange = -3)
+
+	animate(holder_obj, pixel_z = 10, time = 2 SECONDS)
+	animate(holder_obj, pixel_z = 15, time = 1 SECONDS)
+	animate(holder_obj, pixel_z = 10, time = 1 SECONDS)
+	animate(holder_obj, pixel_z = 15, time = 1 SECONDS)
+	animate(holder_obj, pixel_z = 10, time = 1 SECONDS)
```

Comment:
Have you tested that this animation is correct?
I would expect it to look more like this in order to queue
```suggestion
	animate(holder_obj, pixel_z = 10, time = 2 SECONDS)
	animate(pixel_z = 15, time = 1 SECONDS)
	animate(pixel_z = 10, time = 1 SECONDS)
	animate(pixel_z = 15, time = 1 SECONDS)
	animate(pixel_z = 10, time = 1 SECONDS)
```
I suspect the code here might only execute the last animate in this chain?

Reply:
Hard to tell honestly- there were other points where it already wasn't working. There's a chance this never worked as intended, so I don't know what to compare it to